Tasks/task_base.py

```diff
@@ -0,0 +1,68 @@
+# Standard Library
+import abc
+
+class TaskBase():
+    """
+    Responsible for implementing the core logic of the various actions
+    that a drone can take (ex. Movement, Follow, Heal, Decode). Must, at
+    a minimum, implement do(), is_done(), and exit_task().
+    """
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do(self):
```

Comment:
Since `do` is typically a reserved word, can we look into changing `do()` to be `perform()` or `execute()`? That way the naming convention still makes sense.

Reply:
Definitely. I like perform(). It will be easy to swap out the names.